internal/host/plugin/repository_host_catalog.go

```diff
@@ -122,7 +122,14 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
 			if !pluginCalledSuccessfully {
 				plgResp, err = plgClient.OnCreateCatalog(ctx, &plgpb.OnCreateCatalogRequest{Catalog: plgHc})
 				if err != nil {
-					if status.Code(err) != codes.Unimplemented {
+					switch {
```

Comment:
Can you add tests to verify these behave as expected?  For the changes in this file and the repository_host_set file.